backend/substrapp/tasks/tasks.py

```diff
@@ -228,14 +237,24 @@ def prepare_aggregatetuple_input_models(directory, tuple_):
     models = []
 
     for input_model in input_models:
-        proc = Thread(target=fetch_model,
-                      args=(AGGREGATETUPLE_TYPE, authorized_types, input_model, directory))
+        proc = ExceptionThread(target=fetch_model,
+                               args=(AGGREGATETUPLE_TYPE, authorized_types, input_model, directory))
         models.append(proc)
         proc.start()
 
     for proc in models:
         proc.join()
 
+    exceptions = []
+
+    for proc in models:
+        if hasattr(proc, "_exception"):
+            exceptions.append(proc._exception)
+            logger.exception(proc._exception)
+    else:
+        if exceptions:
+            raise Exception(exceptions)
+
```

Comment:
This suggestion goes a bit beyond the purpose of the PR, but what do you think about putting this logic (running threads and catching exceptions) in a separate function and adding a test making sure that we catch all exceptions no matter in which thread they occur ?

Reply:
Yes, see the last commit :)

Reply:
That was fast :)